### PR TITLE
Support PostgresQL and SQLite backends

### PR DESCRIPTION
**Pull request by shpun817**
Issue: [StarfishQL backed by multiple database backends #7](https://github.com/SeaQL/starfish-ql/issues/7)

### CODE REVIEW

**Comment by billy1624** [comment]
Hey @shpun817, this issue seems unresolved, https://github.com/SeaQL/starfish-ql/issues/7#issuecomment-1104789809

**Comment by shpun817** [comment]
[#7](https://github.com/SeaQL/starfish-ql/issues/7#issuecomment-1104789809) has been resolved by batch updating connectivity.

The only issue left now is the non-deterministic behaviour for (possibly some versions of) PostgresQL in the connectivity unit tests. However, the cause might be hard to identity due to non-reproducibility.

File: starfish/starfish-core/src/mutate/edge.rs [hunk 1]
```diff
@@ -393,14 +396,49 @@ impl Mutate {
...
+            return Ok(());
+        }
+
         // map_id_to_ancestors is ready; the sizes of the sets in its values are the compound in_conn
         let cols = [
             NodeIden::Name.into_iden(),
             Alias::new(&format!("{}_{}", relation_name, col_name)).into_iden(),
         ];
+        
+        if cfg!(feature = "sqlx-sqlite") {
```
Comment (billy1624) [inline starfish/starfish-core/src/mutate/edge.rs:409]: We should check if the `DbBackend` of `DbConn` was SQLite or not, instead of feature guarded

Comment (shpun817) [reply starfish/starfish-core/src/mutate/edge.rs:409]: Oh it didn't come to my mind

File: starfish/starfish-core/src/query/mod.rs [hunk 2]
```diff
@@ -400,7 +400,7 @@ impl Query {
             let stmt = sea_query::Query::select()
                 .column(NodeQueryIden::Name)
                 .expr_as(Expr::col(Alias::new(&weight_key)), NodeQueryIden::Weight)
-                .expr_as(Expr::val(Some(0_u64)), NodeQueryIden::Depth)
+                .expr_as(Expr::val(Option::<u32>::None), NodeQueryIden::Depth)
```
Comment (billy1624) [inline starfish/starfish-core/src/query/mod.rs:403]: Why this is changed from `Some(0_u64)` to `Option::<u32>::None`?
Isn't it be changed to `Some(0_u32)`?

Comment (shpun817) [reply starfish/starfish-core/src/query/mod.rs:403]: It was changed from `Some(0_u64)` to `Option::<u64>::None` at some point before (when I implemented counting depth), and then to `Option::<u32>::None` for PostgresQL support specifically.

As the depth is counted with Rust, this value is only a placeholder for querying the nodes. I suppose it doesn't matter practically whether `0` or `None` is used, but `None` seems to fit the meaning of placeholder better?

**Review by billy1624 — APPROVED** [review]
Thanks! @shpun817

**Comment by tyt2y3** [comment]
This is one important values we held at SeaQL: be (water LOL) backend agnostic

**Pull request merged**